tests/test_layabout.py

```diff
@@ -1 +1,327 @@
-import layabout  # noqa: F401
+import os
+from unittest.mock import MagicMock, call
+
+import pytest
+
+from layabout import (
+    Layabout,
+    MissingSlackToken,
+    FailedConnection,
+)
+
+# The likelihood of this ever being a valid Slack token is probably slim.
+TOKEN = 'xoxb-13376661337-DeAdb33Fd15EA53fACef33D1'
+
+
+@pytest.fixture
+def slack_client():
+    def api_call(method):
+        return dict(
+            channels=[{'id': 'D3ADB33F1', 'name': 'general'}],
+            groups=[{'id': '1800IDGAF', 'name': 'super_secret_club'}],
+        )
+
+    slack_client = MagicMock(api_call=api_call)
+    return slack_client
+
+
+def test_layabout_can_register_handlers():
+    """
+    Test that layabout can register Slack API event handlers normally and as a
+    decorator, both with and without keyword arguments.
+    """
+    layabout = Layabout(name='test')
+    kwargs = dict(spam='🍳')
+
+    def handle_hello1(slack, event):
+        pass
+
+    @layabout.handle('hello')
+    def handle_hello2(slack, event):
```

Comment:
Are `hello2` and `hello4` ever used?

Reply:
In this test being decorated is being used.
```python
@layabout.handle('hello')
def handle_hello2(slack, event): ...
```
is meant to verify that nothing explodes when a handler is registered via decorator. Likewise
```python
@layabout.handle('hello', kwargs=kwargs)
def handle_hello4(slack, event): ...
```
is meant to verify that nothing explodes when a handler is registered via decorator with keyword arguments.

I could break all 4 of these cases up into separate tests without issue. Mainly, as long as an exception isn't thrown then these functions ought to have been registered properly and the test should pass.